### PR TITLE
Fix "original work" tag for tag filer

### PR DESCRIPTION
**Pull request by revam**
and modernise _some_ of the code.

It works as expected on my-machine™ from the v3 api.

Haven't tested the the changes for the series search and the v2 filters, but the code looks fine imo.

### CODE REVIEW

File: Shoko.Server/API/v2/Models/common/Filters.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
```
Comment (revam) [inline Shoko.Server/API/v2/Models/common/Filters.cs:1]: Unintended.

File: Shoko.Server/Utilities/TagFilter.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿using System;
+using System;
```
Comment (revam) [inline Shoko.Server/Utilities/TagFilter.cs:1]: Unintended.

**Comment by revam** [comment]
Or so i thought.

**Comment by Cazzar** [comment]
This would need to be heavily tested in v2

I would also ask why are you changing the name and signature of functions?

**Comment by revam** [comment]
The signatures changed because the parameters was not needed in the updated function. The names i can revert if you don't like it.

**Comment by revam** [comment]
Reason; probably needs more testing.

**Comment by revam** [comment]
I modified it to queue the "original work" tag for removal, because we keep it and discard the "new" tag if the "new" tag is not queued for removal.

This time I tested with multiple allow/deny filters and with multiple series (anidb ids: 9556 (anime original), 14477 (novel) and 6246 (manga) ), though i couldn't find a series that's missing a "source"-type tag in my collection to test the edge case where no source tags are present.

**Comment by revam** [comment]
> This would need to be heavily tested in v2
> 
> I would also ask why are you changing the name and signature of functions?

I changed the signatures since i noticed that the two last arguments were never used in the current code-base. The `bool addTags` was always set to true (the default) so it always added the new/renamed tags and the `bool invert` (which was set to false by default) was only used if the tag-list contained one element and we already had an invert flag in the filter so i didn't see the need for it.

**Comment by da3dsoul** [comment]
I think I want to try something. This is also a good class to write unit tests for, since it's pretty well generified already. I'll pull this down after work and mess with it

**Comment by da3dsoul** [comment]
Put into the tagfilter branch. I'll open another PR for that one
